### ISSUE TITLE
Database instances limit

### ISSUE THREAD

**Issue by ubbeK** [Starcounter/Home#467]
Starcounter version: `<2.3.1.8415>`.

### Issue type

- [x] Bug
- [ ] Feature request
- [ ] Suggestion
- [x] Question
- [ ] Cannot reproduce
- [ ] Urgent

### Issue description
Is there a limit in how many Starcounter databases that can be run simultaneously on a server? 

When I start the 16th database, the server crashes.

### Reproduction code snippet

```batch
:: Create databases
staradmin --database=8080 new db DefaultUserHttpPort=8080
staradmin --database=8081 new db DefaultUserHttpPort=8081
staradmin --database=8082 new db DefaultUserHttpPort=8082
staradmin --database=8083 new db DefaultUserHttpPort=8083
staradmin --database=8084 new db DefaultUserHttpPort=8084
staradmin --database=8085 new db DefaultUserHttpPort=8085
staradmin --database=8086 new db DefaultUserHttpPort=8086
staradmin --database=8087 new db DefaultUserHttpPort=8087
staradmin --database=8088 new db DefaultUserHttpPort=8088
staradmin --database=8089 new db DefaultUserHttpPort=8089
staradmin --database=8090 new db DefaultUserHttpPort=8090
staradmin --database=8091 new db DefaultUserHttpPort=8091
staradmin --database=8092 new db DefaultUserHttpPort=8092
staradmin --database=8093 new db DefaultUserHttpPort=8093
staradmin --database=8094 new db DefaultUserHttpPort=8094
staradmin --database=8095 new db DefaultUserHttpPort=8095

:: Start the first 15 database
staradmin --database=8080 start db
staradmin --database=8081 start db
staradmin --database=8082 start db
staradmin --database=8083 start db
staradmin --database=8084 start db
staradmin --database=8085 start db
staradmin --database=8086 start db
staradmin --database=8087 start db
staradmin --database=8088 start db
staradmin --database=8089 start db
staradmin --database=8090 start db
staradmin --database=8091 start db
staradmin --database=8092 start db
staradmin --database=8093 start db
staradmin --database=8094 start db

:: Start the 16th database, the server crashes
staradmin --database=8095 start db
```

### Exception stack trace

```
C:\> staradmin --database=8095 start db
Starting 8095 (starting database ..)                                                                                                                                                                                                                                                                                         
[10:53:58, Starcounter (8095)]

System.Exception: ScErrUnspecified (SCERR999): An unspecified error caused the operation to fail. System.Net.Sockets.SocketException (0x80004005): An existing connection was forcibly closed by the remote host
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
   at Starcounter.NodeTask.PerformSyncRequest() in C:\TeamCity\BuildAgent\work\sc-17027\Level1\src\Starcounter.Internal\Rest\NodeTask.cs:line 514
Version: 2.3.1.8415.
Help page: https://docs.starcounter.io/v/2.3.1/?q=SCERR999.
   at Starcounter.Internal.AppsBootstrapper.RegisterNewCodehostInGateway() in C:\TeamCity\BuildAgent\work\sc-17027\Level1\src\Starcounter.Apps.JsonPatch\AppsBootstrapper.cs:line 64
   at Starcounter.Internal.AppsBootstrapper.InitAppsBootstrapper(Byte numSchedulers, UInt16 defaultUserHttpPort, UInt16 defaultSystemHttpPort, UInt32 sessionTimeoutMinutes, String dbName, Boolean noNetworkGateway) in C:\TeamCity\BuildAgent\work\sc-17027\Level1\src\Starcounter.Apps.JsonPatch\AppsBootstrapper.cs:line 139
   at StarcounterInternal.Bootstrap.Control.Setup(String[] args) in C:\TeamCity\BuildAgent\work\sc-17027\Level1\src\Starcounter.Bootstrap\Control.cs:line 216
   at StarcounterInternal.Bootstrap.Control.Main(String[] args) in C:\TeamCity\BuildAgent\work\sc-17027\Level1\src\Starcounter.Bootstrap\Control.cs:line 55
HResult=-2146233088
HelpLink=https://docs.starcounter.io/v/2.3.1/?q=SCERR999
```
### Event log stack trace
```
Faulting application name: scnetworkgateway.exe, version: 0.0.0.0, time stamp: 0x5ac61add
Faulting module name: ucrtbase.dll, version: 10.0.17763.404, time stamp: 0x490b0aeb
Exception code: 0xc0000409
Fault offset: 0x000000000006e91e
Faulting process id: 0x23b8
Faulting application start time: 0x01d50f0a0eb7309a
Faulting application path: C:\Program Files\Starcounter\scnetworkgateway.exe
Faulting module path: C:\Windows\System32\ucrtbase.dll
Report Id: 3bb34588-2331-473f-92f9-060b5863b3b9
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by bigwad** [comment]
From the database engine perspective - there is a limit of 255 running databases.
On this specific error - it's something inside the network gateway. cc @alemoi

**Comment by alemoi** [comment]
Its just a badly set maximum number of active databases in gateway, a constant.

**Comment by alemoi** [comment]
I have now changed the limit to 128 databases. I think this should be enough.

**Comment by ubbeK** [comment]
> I have now changed the limit to 128 databases. I think this should be enough.

It will, thanks.

**Comment by alemoi** [comment]
@ubbeK good :) Now the question is: would you like me to produce the 2.3.1 new build or 2.3.2 build?

**Comment by ubbeK** [comment]
At the moment, we have set up our CI process so we will never exceed 15 instances on one server. So this issue can be closed but if something changes, we would either want the next Starcounter (2.4, 2.5, 3.0 or whatever it will be called) to be released or a new version of 2.3.1.

In case you guys where wondering why we needed more than 15 simultaneously running instances at one point was because we have a internal server which we use for manual testing of different git branches. I wanted to dedicate 2 instances (ports) per developer and 4 ports for our different branches which are in different release stages, meaning that I needed exactly 16 instances. When I tried it, I simply got hit by this. But we have now changed our git branching strategy and all developers only have one dedicated port each.
